liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -962,7 +963,38 @@ private Xpp3Dom getPluginConfig(Plugin plugin, Xpp3Dom overrides) {
         installFeatureParams.addAll(commonParams);
     }
 
+    private static final Map<String, String> aliasMap;
+    static {
+        Map<String, String>tempMap = new HashMap<String, String>();
+        tempMap.put("runtimeArtifact", "assemblyArtifact");
+        tempMap.put("runtimeArchive", "assemblyArchive");
+        tempMap.put("runtimeInstallDirectory", "assemblyInstallDirectory");
+        tempMap.put("configFile", "serverXmlFile");
+        tempMap.put("serverEnv", "serverEnvFile");
+        tempMap.put("installAppPackages", "deployPackages");
+        aliasMap = Collections.unmodifiableMap(tempMap);
+    }
+
     private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
+        // convert alias parameter key to actual parameter key
+        Xpp3Dom alias;
+        for (String key : aliasMap.keySet()) {
+            alias = config.getChild(key);
+            if (alias != null) {
+                if ("runtimeArtifact".contentEquals(key)) {
+                    Xpp3Dom artifact = new Xpp3Dom(aliasMap.get(key));
+                    for (Xpp3Dom child : alias.getChildren()) {
+                        artifact.addChild(child);
+                    }
+                    config.addChild(artifact);
+                } else {
+                    Element e = (element(name(aliasMap.get(key)), alias.getValue()));
+                    config.addChild(e.toDom());
+                }
```

Comment:
Does it need to remove the old `alias` from `config` after adding the new object?

Reply:
It will be removed later in the method when the non applicable goal parameters are removed 
```
        // strip non applicable parameters
        List<Integer> removeChildren = new ArrayList<Integer>();
        for (int i=0; i<config.getChildCount(); i++) {
            if (!goalParams.contains(config.getChild(i).getName().trim())) {
                removeChildren.add(i);
            }
        }
        Collections.reverse(removeChildren);
        for (int child : removeChildren) {
            config.removeChild(child);
        }
```